src/main/java/com/tlswe/awsmock/ec2/control/MockEC2QueryHandler.java

```diff
@@ -2048,4 +2054,19 @@ private String getBlankResponseXml() {
         }
         return ret;
     }
+
+    /**
+     * Gets the VPC id for a given subnetId.
+     *
+     * @param subnetId The subnet id.
+     * @return The VPC id. Returns null, if no matching subnet is found.
+     */
+    private String getVpcForSubnetId(final String subnetId) {
+        for (MockSubnet subnet : mockSubnetController.describeSubnets()) {
+            if (subnet.getSubnetId().equals(subnetId)) {
+                return subnet.getVpcId();
+            }
+        }
+        return null;
```

Comment:
Since it's always expected that a VPC is tied to a subnet - it's better if we throw an Exception than returning null. It will also be difficult to catch if for some reason we do return null.

Reply:
If we use a default value for not provided subnetIds, then we need here either a check for MOCK_SUBNET_ID where we return MOCK_VPC_ID. Or we just return MOCK_VPC_ID by default. 

`        if (MOCK_SUBNET_ID.equals(subnetId)) {
            return MOCK_VPC_ID;
        }`
Or
`
    private String getVpcIdForSubnet(final String subnetId) {
        for (MockSubnet subnet : mockSubnetController.describeSubnets()) {
            if (subnet.getSubnetId().equals(subnetId)) {
                return subnet.getVpcId();
            }
        }
        return MOCK_VPC_ID;
    }
`
What would be your prefered solution?

Reply:
I feel we can return `MOCK_VPC_ID` if no matching `vpcId` found.